internal/simple_client.go

```diff
@@ -14,6 +15,22 @@ type Client struct {
 	QueueUrl string
 }
 
+func extractAccountIDFromSQSQueueURL(queueURL string) (string, error) {
+	// Define the regular expression pattern to match the AWS account ID
+	pattern := `https://sqs\.[a-zA-Z0-9-]+\.amazonaws\.com/(\d+)/`
+
+	// Compile the regular expression
+	re := regexp.MustCompile(pattern)
+
+	// Find matches in the queue URL
+	matches := re.FindStringSubmatch(queueURL)
+
+	// The AWS account ID is captured in the first submatch
+	accountId := matches[1]
+
+	return accountId, nil
+}
+
```

Comment:
Please add a few test cases to the table-driven `func Test_NewClient` test in [simple_client_test.go](https://github.com/engelmi/go-sqs/blob/main/test/internal/simple_client_test.go) to verify that this doesn't break anything.